### ISSUE TITLE
Make legend an interactive filter

### ISSUE THREAD

**Issue by natmey** [newamericafoundation/drones#43]
The new legend is awesome.

I would like to make it interactive. I'm thinking something like the way Google calendars work, where when you click on a calendar color square, it toggles that calendar on your view.

This is to replicate the existing functionality of the filter checkboxes, so that filtering can happen in multiple ways.

**Comment by kakaes** [comment]
Agree, that'd be lovely if it's simple enough for Jeff to implement?

**Comment by sajacy** [comment]
It definitely sounds nice - but it's a little more involved than meets the eye, because it's not really a 1:1 mapping with the filters - there's some relatively heavy-handed logic that would need to make it work nicely for the user, i.e. a user would need to toggle everything else off to select only one kind (like the way the filters work), and we would need to ignore clicks on the "multiple missions" legend item (or retrofit the filtering code to detect multiple-mission flights), and synchronize the filter menu selections with the legend selections. (which are essentially inverts of each other).

**Comment by kakaes** [comment]
That sounds to me like more hassle than it's worth, to duplicate functionality that's already there, in a slightly different UI. But nice idea--thanks for parsing it Jeff!
